pkg/apis/ibmcloud/v1alpha1/ibmcloudmachineproviderspec_types.go

```diff
@@ -43,6 +44,9 @@ type IbmcloudMachineProviderSpec struct {
 	LocalDiskFlag bool `json:"localDiskFlag,omitempty"`
 
 	HourlyBillingFlag bool `json:"hourlyBillingFlag,omitempty"`
+
+	UserDataSecret *corev1.SecretReference `json:"userDataSecret,omitempty"`
```

Comment:
Please refer to https://github.com/multicloudlab/cluster-api-provider-ibmcloud/blob/master/CONTRIBUTING.md#pre-check-before-submitting-a-pr for some pre-check.

Reply:
I met it before. Just forgot to commit this latest fix. I will push the updated version.